redis/cluster.py

```diff
@@ -157,6 +157,7 @@ def parse_cluster_myshardid(resp, **options):
     "ssl_cert_reqs",
     "ssl_keyfile",
     "ssl_password",
+    "timeout",  # added timeout to support injecting BlockingConnectionPool timeout
```

Comment:
is this adjust the timeout for how long redis commands can take?

Reply:
nope, it's timeout for fetching a connection from connection pool. There is no configuration to limit the redis operation duration

Reply:
got it